libproject/devstart.rb

```diff
@@ -0,0 +1,90 @@
+require "optparse"
+require "fileutils"
+require_relative "../aou-utils/utils/common"
+require_relative "../aou-utils/serviceaccounts"
+require_relative "../aou-utils/workbench"
+require_relative "../aou-utils/swagger"
+
+# Update this whenever we want to generate libraries for the latest version of the workbench API.
+SWAGGER_SPEC = 'https://raw.githubusercontent.com/all-of-us/workbench/fac5e45d578d1e5f6d735824a89b1d4f5a8f8f37/api/src/main/resources/workbench.yaml'
+TEST_PROJECT = "all-of-us-workbench-test"
+
+def swagger_regen()
+  Workbench::Swagger.download_swagger_codegen_cli
+
+  common = Common.new
+  common.run_inline %W{
+      java -jar #{Workbench::Swagger::SWAGGER_CODEGEN_CLI_JAR}
+      generate --lang python --input-spec #{SWAGGER_SPEC} --output py/tmp}
+  file_opts = {:verbose => true}
+  FileUtils.rm_rf('py/aou_workbench_client/swagger_client', file_opts)
+  FileUtils.rm_rf('py/swagger_docs', file_opts)
+  FileUtils.mv('py/tmp/swagger_client', 'py/aou_workbench_client/swagger_client', file_opts)
```

Comment:
Why the tmpdir shuffling? Is this to whitelist certain files? Note that you can use a swagger ignore file to blacklist, e.g.:

https://github.com/DataBiosphere/job-manager/blob/master/servers/dsub/.swagger-codegen-ignore

Reply:
As discussed over Slack: we want to be able to put swagger_docs outside of the aou_workbench_client dir, and delete both swagger_client and swagger_docs explicitly before regenerating (to make sure we're not leaving around any old files.) The tmp approach works well for that.

I added .swagger-codegen-ignore at the root and copied it into py/tmp before generating just to save the trouble of generating files we're going to ignore.